Private key loaded via ENGINE_load_private_key is not usable.

Due to "pkey->pmeth_engine" becomes NULL,
pkey loses its reference to methods provided by engine.
Issue has been reported to OpenSSL commnunity.
https://github.com/openssl/openssl/issues/22945.
It is workaround patch for webOS Keymanager engine,
and will be temprorarily valid until official fix comes up.

Upstream-Status: Pending

===================================================================
--- openssl-3.0.12.orig/crypto/engine/eng_pkey.c	2023-10-24 22:48:41.000000000 +0900
+++ openssl-3.0.12/crypto/engine/eng_pkey.c	2023-12-05 17:25:44.150103409 +0900
@@ -79,6 +79,10 @@
         ERR_raise(ERR_LIB_ENGINE, ENGINE_R_FAILED_LOADING_PRIVATE_KEY);
         return NULL;
     }
+    if (e->id != NULL && !strncmp(e->id, "libkm_engine", strlen("libkm_engine"))) {
+        return pkey;
+    }
+
     /* We enforce check for legacy key */
     switch (EVP_PKEY_get_id(pkey)) {
     case EVP_PKEY_RSA:
